frontend/src/services/QueryServiceImpl.ts

```diff
@@ -1,94 +1,48 @@
 import { CancellationToken } from '../utils/CancellationToken';
-import nop from '../utils/nop';
+import HttpService from './http';
 import QueryService from './QueryService';
-import { QueryServiceOptions } from './QueryServiceOptions';
-
-/**
- * Builds the query service options.
- * @param options The partial options that very specified externally.
- * @returns The built options.
- */
-function buildOptions(
-  options: Partial<QueryServiceOptions>
-): QueryServiceOptions {
-  let { backendBaseUri } = options;
-
-  // As per the spec of QueryServiceOptions, we use the base URI of the source the frontend
-  // was loaded from, if no backend base uri was specified in the options.
-  if (!backendBaseUri) {
-    const { location } = window;
-    backendBaseUri = `${location.protocol}//${location.host}/${
-      location.pathname.split('/')[1]
-    }`;
-  }
-
-  return { backendBaseUri };
-}
 
 /**
  * The implementation of query service that performs query requests
  * via the backend.
  */
 @injectable()
 export default class QueryServiceImpl extends QueryService {
-  private readonly options: QueryServiceOptions;
-
-  public constructor(options: Partial<QueryServiceOptions> = {}) {
-    super();
-
-    this.options = buildOptions(options);
-  }
+  @inject(HttpService)
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  private readonly http: HttpService = null!;
 
   public queryAll(
     query?: LimitQuery,
     cancellation?: CancellationToken
   ): Promise<QueryResult> {
-    const url = `${this.options.backendBaseUri}/queryAll`;
-
-    return new Promise<QueryResult>((resolve, reject) => {
-      if (cancellation?.isCancellationRequested) {
-        reject(new CancellationError());
-      }
-
-      const request = new XMLHttpRequest();
-      request.open('post', url, true);
-      request.setRequestHeader('Content-Type', 'application/json');
-
-      let cancellationUnsubscribe = nop;
-
-      if (cancellation) {
-        cancellationUnsubscribe = cancellation.subscribe(() => request.abort());
-      }
+    const url = `/queryAll`;
 
-      request.onload = () => {
-        cancellationUnsubscribe();
-        if (request.status >= 200 && request.status <= 299) {
-          resolve(JSON.parse(request.response)); // TODO: Validate the response?
-        } else {
-          reject(Error(request.statusText));
-        }
-      };
-
-      request.onerror = () => {
-        cancellationUnsubscribe();
-        if (cancellation?.isCancellationRequested) {
-          reject(new CancellationError());
-        } else {
-          reject(Error('A network error occured.'));
-        }
-      };
-
-      request.onabort = () => {
-        cancellationUnsubscribe();
-        reject(new CancellationError());
-      };
+    return this.http.post<QueryResult>(url, query, cancellation);
+  }
 
-      request.send(JSON.stringify(query ?? {}));
-    });
+  public fullTextSearch(
```

Comment:
Why isn't this in a dedicated service? In the backend it is not part of this service and the HTTP API is on the subpath `search/` while all urls this service maes available is on the root path.

Reply:
See comment above